src/compiler/factory/node_factory.rs

```diff
@@ -108,24 +108,41 @@ impl NodeFactory {
         &self,
         base_factory: &TBaseNodeFactory,
         kind: SyntaxKind,
-        value: &str,
+        value: String,
     ) -> BaseLiteralLikeNode {
         let node = self.create_base_token(base_factory, kind);
-        BaseLiteralLikeNode {
-            _node: node,
-            text: value.to_string(),
-        }
+        BaseLiteralLikeNode::new(node, value)
     }
 
     pub fn create_numeric_literal<TBaseNodeFactory: BaseNodeFactory>(
         &self,
         base_factory: &TBaseNodeFactory,
-        value: &str,
+        value: String,
```

Comment:
I'd generally push the conversion of `&str` into `String` until the last possible place, but maybe that doesn't make sense here.